spec/models/user_spec.rb

```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+describe User, type: :model do
+  context "with no username" do
+    it "is invalid" do
+      user = User.new
+      expect(user).to_not be_valid
```

Comment:
Might be good to have a more specific assertion here, can you think of any ways to assert the user is not valid for the correct reason?

Reply:
A better reason for a user not to be valid, would be if no password was passed in as an argument.

Reply:
Sorry, maybe I didn't express myself well in first question. 

What I was trying to say is a user could be invalid for a number of reasons, i.e. no password, username, or any other model validations you make have defined. 

I thought it would be cool if you could do something like `expect(user.errors[:foo]).to eql("funny error msg")`

Reply:
I see, yes!
Being more specific, which ensures that the failure is due to what we are testing for specifically in this test and not because of some other possibly non related reasons.

Reply:
Something more like this?
```ruby
user = User.new(username: 'testUser')
expect(user.valid?).to be(false)
expect(user.errors.messages[:password_digest]).to include("can't be blank")
```